go-controller/pkg/ovn/egressgw.go

```diff
@@ -349,7 +349,13 @@ func (oc *Controller) addHybridRoutePolicyForPod(podIP net.IP, node string) erro
 		if err != nil {
 			return fmt.Errorf("failed to parse gateway router join interface IP: %s, err: %v", grJoinIP, err)
 		}
+		var matchDst string
+		for _, clusterSubnet := range config.Default.ClusterSubnets {
+			matchDst += fmt.Sprintf(" && %s.dst != %s", l3Prefix, clusterSubnet.CIDR)
```

Comment:
we don' t have any test that checks these expressions?

Reply:
We have e2e control plane tests that verify egress gw works. However those test do not validate pod to service (regular k8s) traffic on egress gw pods. We should add a test for this @nerdalert

Reply:
Actually, I should take that.
I pulled that work from @nerdalert .